docs(databases): Add permissions how-to page int-add-databases (#500)
scaleway/docs-content: storage/database/how-to/manage-permissions.mdx

---
meta:
  title: How to manage user permissions for Managed Databases
  description: This page explains how to manage user permissions on a Managed Database
content:
  h1: How to manage user permissions for Managed Databases
  paragraph: This page explains how to manage user permissions on a Managed Database
dates:

>   validation: 2022-03-30

Reviewer: ```suggestion
  validation: 2022-03-25
```

Author: Fixed!

  posted: 2022-03-25
---

> Permissions define the actions a user can perform on a logical database. By default, no permission are given to a newly created user, except for the initial user created during the instance configuration (which is granted all permissions). 

Reviewer 2: ```suggestion
Permissions define the actions a user can perform on a logical database. A database user is generated automatically when you create your Database Instance and this user is attributed all permissions. However, newly created users have no permissions by default. 
```

But Managed Databases allows you to fine tune this to fit your needs and ensure your data security.

<Message type="requirement">

- You have an account and are logged into the [Scaleway console](https://console.scaleway.com)
- You have a [Managed Database](/storage/database/concepts/#managed-database/)

</Message>

## Set permissions for an existing user

1. Click **Database** on the side menu. A list of your Database Instances displays.
2. Click the database name or <Icon name="more" /> > **More info** to access the Database information page.
3. Go to the **Users** tab. A list of users displays.
4. Click <Icon name="more" /> > Update permissions to open the user permissions panel.
5. Select the appropriate permissions for each databases.

  <Message type="note">

  Three types of permissions are available:
    - **None:** No access to the database
    - **Read:** Allow users to read tables and fields in a database
    - **Write:** Allow users to write content in databases

  </Message>

> 7. Click **Update user** to confirm. 

Reviewer 2: ```suggestion
6. Click **Update user** to confirm. 
```

## Set permissions for a new user

During user creation, the permissions panel can be accessed through the top tab menu. 
For more information see the [add users how to](/database/how-to/add-users/)

<Message type="note">

> Permissions are applied on existing objects **at the moment the configuration is set**, and not applied to any object created afterward. 

Reviewer 2: ```suggestion
Permissions are applied on existing items **at the moment the configuration is set**, and not applied to any items created afterward. 
```

> Permissions are automatically set to "**Custom**" when the database structure was modified after the permission was set. "**Custom**" means the initial permission is partially applied.

Reviewer 2: **Custom** means the initial permissions are partially applied. -- What does it mean to have the permissions partially applied? What happens to the other part?
```suggestion
When you modify your database configuration, your configured permissions are automatically set to **Custom**. **Custom** means the initial permissions are partially applied.
```

Author: :wave: 
It means the user does not have the permission applied to the new items. The user will still have the initial permission on the old items and But the `None` permission on the new ones (like if no permission was set).

Reviewer 2: Ok, got it. Thank you! I added to my suggestion accordingly 🙏🏻
